Documentation for CC deployment (#23)
kyguy/strimzi-kafka-operator: documentation/modules/cruise-control/ref-cruise-control-configuration.adoc

// Module included in the following assemblies:
//
// assembly-cruise-control-concepts.adoc

[id='ref-cruise-control-configuration-{context}']
= Cruise Control configuration

> The `config` property in `Kafka.spec.cruiseControl` contains configuration options as keys with values in one of the following JSON types:

Reviewer: with values *as* one of

> The `config` property in `Kafka.spec.cruiseControl` contains Cruise Control configuration options as keys with values in one of the following JSON types:

Reviewer: Here we are referring to "configuration options", instead of "properties". I think we need to consistent in what we call these things.

Reviewer: FWIW I think using configuration options throughout would be better. But I may just be being needlessly picky.

Author: `Kafka.spec.cruiseControl.config` is a property that holds the the configuration options for Cruise Control. `Kafka.spec.cruiseControl` is a property that has other properties for configuring the Cruise Control deployment

> The `config` property in `Kafka.spec.cruiseControl` contains Cruise Control configuration options as keys with values in one of the following JSON types:

Reviewer 2: ```suggestion
The `config` property in `Kafka.spec.cruiseControl` contains configuration options as keys with values in one of the following JSON types:
```

* String
* Number
* Boolean

> You can specify and configure all of the options in the "Configurations" section of the {CruiseControlConfig} apart from those managed directly by {ProductName}.

Reviewer 2: ```suggestion
You can specify and configure all the options listed in the "Configurations" section of the {CruiseControlConfigDocs}, apart from those managed directly by {ProductName}.
```

> Specifically, you are prevented from modifying all configuration options with keys equal to or starting with one of the following strings:

Reviewer 2: ```suggestion
Specifically, you *cannot* modify configuration options with keys equal to or starting with one of the following strings:
```

* `bootstrap.servers`
* `zookeeper.`
* `ssl.`
* `security.`
* `failed.brokers.zk.path`
* `webserver.http.port`
* `webserver.http.address`
* `webserver.api.urlprefix`
* `metric.reporter.sampler.bootstrap.servers`
* `metric.reporter.topic`
* `metric.reporter.topic.pattern`
* `partition.metric.sample.store.topic`
* `broker.metric.sample.store.topic`
* `capacity.config.file`
* `skip.sample.store.topic.rack.awareness.check`
* `cruise.control.metrics.topic`
* `sasl.`

> If the `config` property specifies a restricted option, it is ignored and a warning message is printed to the Cluster Operator log file.

Reviewer 2: ```suggestion
If restricted options are specified, they are ignored and a warning message is printed to the Cluster Operator log file.
```

> All other supported options are passed to Kafka.

Reviewer: `:%s/Kafka/Cruise Control`

> All other supported options are passed to Cruise Control.

Reviewer 2: ```suggestion
All supported options are passed to Cruise Control.
```

.An example Cruise Control configuration
[source,yaml,subs="attributes+"]
----
apiVersion: {KafkaApiVersion}
kind: Kafka
metadata:
  name: my-cluster
spec:
  # ...
  cruiseControl:
    # ...
    config:
      default.goals: >

>          com.linkedin.kafka.cruisecontrol.analyzer.goals.RackAwareGoal,

Reviewer: See previous comment about valid yaml lists.

Author: See above
